server/src/main/java/org/opensearch/telemetry/tracing/TracerFactory.java

```diff
@@ -30,21 +30,20 @@ public class TracerFactory implements Closeable {
     private static final Logger logger = LogManager.getLogger(TracerFactory.class);
 
     private final TelemetrySettings telemetrySettings;
-    private final Tracer defaultTracer;
+    private final Tracer tracer;
 
     public TracerFactory(TelemetrySettings telemetrySettings, Optional<Telemetry> telemetry, ThreadContext threadContext) {
```

Comment:
I think we don't even need tracer factory now, we could inject tracer directly instead

Reply:
You mean in instrumentation code (as i too [commented](https://github.com/opensearch-project/OpenSearch/pull/8408#discussion_r1258250390) in other PR) or in general(remove tracer factory altogether)?

Reply:
I think right now tracer factory is not needed as injectable component: it is direct pass-through to the tracer.

Reply:
Yeah I am also in favour to use `Tracer.java` as injectable component in instrumentation code.
@Gaganjuneja

Reply:
Thanks @suranjay , we could address that next